### ISSUE TITLE
Test avoiding `InExactError` by using `TransformObservationModel`

### ISSUE THREAD

**Issue by SamuelBrand1** [CDCgov/Rt-without-renewal#561]
As a reminder this is caused by the inference expecting `Int` valued returns but instead getting caught by a `BigInt`. This is possible with latent processes in the log domain which gets `exp` transformed and then linked with a neg binomial or Poisson distribution e.g.

```julia
julia> typemax(Int) |> log
# 43.66827237527655
```

We've tried a few approaches to stop this eg #492 . 

This issue is about trying a `TransformObservationModel` composition with the usual obs processes to implement:

```math
x \to - \log ( e^{-C} + e^{-x})
```

Where $C$ is a large float. In julia this is 

```julia
using LogExpFunctions
const C = some_big_number
x -> -logaddexp(-C, -x)
```

**Comment by SamuelBrand1** [comment]
The point is that its smooth at the transition point but also effectively linear for most of the domain of relevance.

**Comment by SamuelBrand1** [comment]
Closed by #562
